core-v1/src/main/scala/data/Fact.scala

```diff
@@ -91,10 +91,7 @@ object TypedFact {
     evidence: Evidence,
   ): TypedFact[A] with DerivedFact = DerivedFactOfType(typeInfo, value, evidence)
 
-  def unapply[A](fact: TypedFact[A]): Some[(FactType[A], A)] = fact match {
-    case SourceFactOfType(typeInfo, value) => Some((typeInfo, value))
-    case DerivedFactOfType(typeInfo, value, _) => Some((typeInfo, value))
-  }
+  def unapply[A](fact: TypedFact[A]): Some[(FactType[A], A)] = Some((fact.typeInfo, fact.value))
```

Comment:
I don't know why this wasn't a compiler error... `TypeFact` was not `sealed`...

Reply:
Hmm... it looks like the Scala compiler *only* checks for exhaustivity of `sealed` traits / classes. Removing the `sealed` qualifier broke the `match` statement AND simultaneously prevented the compiler from flagging it as an error... that kinda sucks. I feel like the compiler should always assume in-exhaustivity for any unsealed (i.e. open) type without a `case _`.